manifests/install.pp

```diff
@@ -23,6 +23,13 @@
     proxy_type   => $duo_authproxy::proxy_type,
   }
 
+  -> exec { 'duoauthproxy-move':
```

Comment:
I'm curious about this. Can you explain a bit?

Reply:
Sure. The Duo Authentication Proxy version ```2.7.0``` when downloaded and extracted, the filename stays the same(for example: ```duoauthproxy-2.7.0-src.tgz``` to ```duoauthproxy-2.7.0-src``` ) so the way you originally have the module works with no problem. However, starting with the Duo Authentication Proxy version 2.10.0 when downloaded and extracted, the filename doesn't stay the same(for example:  ```duoauthproxy-2.10.0-src.tgz``` to ```duoauthproxy-2.10.0-2c96da5-src```). This naming convention also changed with the most recent version ```3.0.0``` (for example: ```duoauthproxy-3.0.0-src.tgz``` to ```duoauthproxy-3.0.0--src```)
The naming convention change with the Duo Authentication Proxy versions caused ```exec { 'duoauthproxy-make':``` and ```exec { 'duoauthproxy-install':``` to break because for the ```cwd``` parameter in those two execs for ```/tmp/duoauthproxy-${duo_authproxy::version}-src```, could not be executed because it tries to execute something like this ```/tmp/duoauthproxy-2.10.0``` instead of  ```/tmp/duoauthproxy-2.10.0-2c96da5-src``` but since there was no ```/tmp/duoauthproxy-2.10.0-2c96da5-src``` puppet could not execute those two execs. 

So all that the ```exec { 'duoauthproxy-move':``` does is, once the the Duo Authentication Proxy is downloaded and extracted, it moves the extracted file to the right name so that ```exec { 'duoauthproxy-make':``` and ```exec { 'duoauthproxy-install':``` can execute. So regardless of what the extracted filename is, the ```exec { 'duoauthproxy-move':``` always ensures that its renamed appropriately (so  for example: ```duoauthproxy-2.10.0-2c96da5-src``` would be renamed to ```duoauthproxy-2.10.0-src```) or (```duoauthproxy-3.0.0--src``` would be renamed to ```duoauthproxy-3.0.0-src```)

You can validate what I have above with the commands below on your linux box and look at the file and directory names:

```
wget https://dl.duosecurity.com/duoauthproxy-2.7.0-src.tgz && tar xzf duoauthproxy-2.7.0-src.tgz
wget https://dl.duosecurity.com/duoauthproxy-2.10.0-src.tgz && tar xzf duoauthproxy-2.10.0-src.tgz
wget https://dl.duosecurity.com/duoauthproxy-3.0.0-src.tgz && tar xzf duoauthproxy-3.0.0-src.tgz
```
Another way to also validate this would be to try changing the version to anything from 2.10.0 and above in your module and then run ```puppet agent -t``` which should give you the same error I mentioned.

Hope this clarifies things a bit. Let me know if you have anymore questions or need further clarification.

Reply:
@edestecd Any update on this?

Reply:
sorry I'll try and get to this sometime this week. I have lots of things due at work

Reply:
Thank you very much. Let me know if you have any questions